src/QsCompiler/CompilationManager/FileContentManager.cs

```diff
@@ -857,22 +857,25 @@ internal void PushChange(TextDocumentContentChangeEvent change, out bool publish
             publishDiagnostics = true;
             if (count == 1 && line == start)
             {
-                // TODO: If the change contains characters that are part of a symbol, we should delay sending
-                // diagnostics until the user has a chance to finish typing, either by queuing the update here or
-                // queuing the diagnostics before they're sent to the client.
-                this.UnprocessedUpdates.Enqueue(change);
                 var trimmedText = change.Text.TrimStart(); // tabs etc inserted by the editor come squashed together with the next inserted character
                 if (change.Text == String.Empty || 
                     trimmedText == "{" || trimmedText == "\"" || // let's not immediately trigger an update for these, hoping the matching one will come right after
                     trimmedText == "\\" || trimmedText == "/") // ... and the same here
                 {
-                    this.Timer.Start(); // we can simply queue this update - no need to actually execute it
+                    // we can simply queue this update - no need to actually execute it
+                    this.UnprocessedUpdates.Enqueue(change);
+                    this.Timer.Start();
                     publishDiagnostics = false;
                     return;
                 }
-                this.Update(); // update only the currently queued changes
```

Comment:
A tiny difference, but if you can call Update() instead of Update(change), you are saving one iteration of the scope tracking update.